src/js/components/Modal/Modal.tsx

```diff
@@ -1,43 +1,57 @@
 import * as React from 'react';
-import { Button } from '../Button';
+import { Icon } from '../Icon';
 
 interface Props {
-  heading: string;
+  heading: React.ReactNode;
   show: boolean;
-  handleCTA: any;
-  actionable?: boolean;
-  ctaLabel?: string;
   children: React.ReactNode;
+  showClose?: boolean;
+  allowOverlayClose?: boolean;
+  style?: React.CSSProperties;
+  closeFn?: () => void;
 }
 
 const Modal: React.FC<Props> = ({
   heading,
   show,
-  handleCTA,
-  actionable,
-  ctaLabel,
-  children
+  children,
+  showClose,
+  closeFn,
+  allowOverlayClose,
+  style
 }: Props) => {
+
+  React.useEffect(() => {
+    if (allowOverlayClose) {
+      const toggleModal = (event: any) => {
+        if (!document.querySelector('.Modal-content').contains(event.target)) {
+          closeFn?.();
+        }
+      }
+
+      document.body.addEventListener('click', toggleModal, { capture: true });
+
+      return () => {
+        document.body.removeEventListener('click', toggleModal);
+      }
+    }
+  },[]);
+
   return (
     <div className={`Modal ${show ? 'Modal--open' : 'Modal--closed'}`}>
-      <div className='Modal-content'>
-        <div className='Modal-header'>
-          <h2 style={{ marginTop: '20px' }}>{heading}</h2>
+      <div className='Modal-content u-posRelative' style={{ width: '50%', ...(style || {})}}>
+        <div className='Modal-header u-flex u-flexJustifyBetween'>
+          <div className="u-sizeFill">
+            <h2>{heading}</h2>
+          </div>
+          { showClose &&
+            <div className="Modal-close">
```

Comment:
Let's stick to using 1 type of quote. Single quotes are usually recommended. Sorry to nitpick here there are other cases up above where we are using double quotes again